phpBB/includes/ucp/ucp_register.php

```diff
@@ -457,30 +457,26 @@ function main($id, $mode)
 
 				if ($config['email_enable'])
 				{
-					include_once($phpbb_root_path . 'includes/functions_messenger.' . $phpEx);
-
-					$messenger = new messenger(false);
-
-					$messenger->template($email_template, $data['lang']);
-
-					$messenger->to($data['email'], $data['username']);
-
-					$messenger->anti_abuse_headers($config, $user);
-
-					$messenger->assign_vars(array(
+					$messenger = $phpbb_container->get('messenger.method_collection');
+					$email = $messenger->offsetGet('messenger.method.email');
```

Comment:
I think there is more places like this and you shouldn't need to go through the method collection if you just get the email method.

Reply:
Also, I'd recommend going for clearer names, e.g. $email_method instead of just $email